flask_peewee/rest/__init__.py

```diff
@@ -770,6 +795,46 @@ def delete(self, obj):
         res = obj.delete_instance(recursive=self.delete_recursive)
         return self.response({'deleted': res})
 
+    def json_value(self, obj, field_name, path):
+        value = getattr(obj, field_name)
+        for key in path:
+            if not isinstance(value, dict) or key not in value:
+                value = None
+                break
+            value = value[key]
+        return self.response(value)
+
+    def json_edit(self, obj, field_name, path):
+        value = getattr(obj, field_name)
```

Comment:
Please add a test for this to confirm, but I think you probably need something along the lines of:

 ```python
value = getattr(obj, field_name) or {}
```

to cover that case that someone `PUT`s to, say, `foo/123/a_field/a/b/c` when `foo.a_field` is currently `None`.

Reply:
Turns out that you can't set a JSONB field to null, only to JSON null, and peewee seems to have a bug doing this via the orm. Will assume all JSONB fields are objects for now.

Reply:
TIL and bummer on the bug. We should file a peewee issue for the bug. Happy to do that if you don't have time.